core/src/util/mod.rs

```diff
@@ -24,6 +28,52 @@ lazy_static! {
     static ref NON_ESCAPED_PERCENT: FancyRegex = FancyRegex::new(r"(?<!\\\\)%").unwrap();
 }
 
+pub(crate) fn handle_libmongosqltranslate_command(command: Document) -> Result<Document> {
+    let library = get_mongosqltranslate_library().expect("libmongosqltranslate is not available.");
+
+    let run_command_function: Symbol<
+        'static,
+        unsafe extern "C" fn(LibmongosqltranslateDataIO) -> LibmongosqltranslateDataIO,
+    > = unsafe { library.get(b"runCommand") }.expect("Failed to load runCommand symbol.");
+
+    let command_type = command.get_str("command").unwrap();
+
+    let command_bytes_vec =
+        bson::to_vec(&command).expect("Failed to serialize Document into BSON bytes.");
+
+    let command_bytes_length = command_bytes_vec.len();
+
+    let command_bytes_capacity = command_bytes_vec.capacity();
+
+    let libmongosqltranslate_command = LibmongosqltranslateDataIO {
+        data: Box::into_raw(command_bytes_vec.into_boxed_slice()).cast(),
+        length: command_bytes_length,
+        capacity: command_bytes_capacity,
+    };
+
+    let decomposed_returned_doc = unsafe { run_command_function(libmongosqltranslate_command) };
+
+    let returned_doc: Document = unsafe {
+        bson::from_slice(
+            Vec::from_raw_parts(
+                decomposed_returned_doc.data.cast_mut(),
+                decomposed_returned_doc.length,
+                decomposed_returned_doc.capacity,
+            )
+            .as_slice(),
```

Comment:
Since we're deserializing into a `Document`, `serde_stacker` can't be used. The only other way I could find to do this would be to use [from_reader](https://docs.rs/bson/2.11.0/bson/document/struct.Document.html#method.from_reader).